File: daily/2026/3474_Lexicographically_Smallest_Generated_String.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    string generateString(string str1, string str2) {
        int n = str1.length();
        int m = str2.length();
        string ans(n + m - 1, 'a');
        vector<bool> fixed(n + m - 1, false);
        for (int i = 0; i < n; i++){
            if (str1[i] == 'T'){
                for (int j = 0; j < m; j++){
                    if (fixed[i + j] && ans[i + j] != str2[j]) return "";
                    ans[i + j] = str2[j];
                    fixed[i + j] = true;
                }
            }
        }
        for (int i = 0; i < n; i++){
            if (str1[i] == 'F'){
                bool different = false;
                int lastest = -1;
                for (int j = i + m - 1; j >= i; j--){
                    if (ans[j] != str2[j - i]) different = true;
                    if (lastest == -1 && !fixed[j]) lastest = j;
                }
                if (different) continue;
                if (lastest == -1) return "";
                ans[lastest] = 'b';
            }
        }
        return ans;
    }
};

int main(){
    string str1 = "TFTF";
    string str2 = "ab";
    // string str1 = "TFTF";
    // string str2 = "abc";
    // string str1 = "F";
    // string str2 = "d";
    Solution S;
    cout << S.generateString(str1, str2) << endl;
    return 0;
}

File: daily_python/3474_Lexicographically_Smallest_Generated_String.py
import os
import sys
from typing import List

class Solution:
    def generateString(self, str1: str, str2: str) -> str:
        n, m = len(str1), len(str2)
        ans = ['a'] * (n + m - 1)
        fixed = [False] * (n + m - 1)
        for i in range(n):
            if str1[i] == 'T':
                for j in range(m):
                    if fixed[i + j] and ans[i + j] != str2[j]:
                        return ""
                    fixed[i + j] = True
                    ans[i + j] = str2[j]
        for i in range(n):
            if str1[i] == 'F':
                is_diff = False
                idx = -1
                for j in range(m - 1, -1, -1):
                    if ans[i + j] != str2[j]:
                        is_diff = True
                    if not fixed[i + j] and idx == -1:
                        idx = i + j
                if is_diff:
                    continue
                if idx == -1:
                    return ""
                ans[idx] = 'b'
        return "".join(ans)

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            str1 = lines[0].split('=')[1].strip()[1:-2]
            str2 = lines[1].split('=')[1].strip()[1:-2]
            dataset.append((str1, str2))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for str1, str2 in dataset:
        results.append(solution.generateString(str1, str2))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')